pkg/database/database.go

```diff
@@ -107,39 +114,71 @@ func (db *Database) QueryNamespacedResourceByName(ctx context.Context, kind, api
 	}
 }
 
-func (db *Database) performResourceQuery(ctx context.Context, query string, args ...string) ([]*unstructured.Unstructured, string, string, error) {
-	castedArgs := make([]interface{}, len(args))
-	for i, v := range args {
-		castedArgs[i] = v
-	}
+func (db *Database) QueryLogURLs(ctx context.Context, kind, apiVersion, namespace, name string) ([]string, error) {
 
-	rows, err := db.db.QueryContext(ctx, query, castedArgs...)
+	strQueryPerformer := newQueryPerformer[string](ctx, db.db)
+	if kind == "Pod" {
+		return strQueryPerformer.performQuery(db.info.GetLogURLsByPodNameSQL(), apiVersion, namespace, name)
+	}
+	var uuids []string
+	var uuid string
+	err := db.db.QueryRowContext(ctx, db.info.GetUUIDSQL(), kind, apiVersion, namespace, name).Scan(&uuid)
 	if err != nil {
-		return nil, "", "", err
+		return nil, err
 	}
-	defer func(rows *sql.Rows) {
-		err = rows.Close()
-	}(rows)
+	uuids = append(uuids, uuid)
+	var podUuids []string
+	type uuidKind struct {
+		Uuid string `json:"uuid"`
+		Kind string `json:"kind"`
+	}
+	i := len(uuids)
+	for i > 0 {
+		parsedRows, err := newQueryPerformer[uuidKind](ctx, db.db).performQueryWithArray(db.info.GetOwnedResourcesSQL(), uuids)
+		if err != nil {
+			return nil, err
+		}
+		uuids = make([]string, 0)
+		for _, row := range parsedRows {
+			if row.Kind == "Pod" {
+				podUuids = append(podUuids, row.Uuid)
+			} else {
+				uuids = append(uuids, row.Uuid)
+			}
+		}
+		i = len(uuids)
+	}
```

Comment:
I think the looping logic could be made more clear by having it all in the `for` line

```suggestion
	for i:= len(uuids); i > 0; i = len(uuids) {
		parsedRows, err := newQueryPerformer[uuidKind](ctx, db.db).performQueryWithArray(db.info.GetOwnedResourcesSQL(), uuids)
		if err != nil {
			return nil, err
		}
		uuids = make([]string, 0)
		for _, row := range parsedRows {
			if row.Kind == "Pod" {
				podUuids = append(podUuids, row.Uuid)
			} else {
				uuids = append(uuids, row.Uuid)
			}
		}
	}
```

Reply:
great point, as I was thinking of a "while" translation I didn't thought of adding the initialize and the "post-loop" change on the `for`.